File: samples/AesDemo-dotNet/CipherStore.cs
using System;
using System.IO;

namespace AesInteropDemo
{

    public class CipherData
    {
        public byte[] InitializationVector;
        public String Passphrase;
        public byte[] Salt;
        public byte[] CipherText;

        public int GetLength()
        {
            return InitializationVector.Length +
                Passphrase.Length +
                Salt.Length +
                CipherText.Length;
        }
    }

    /// <summary>
    /// Summary description for CipherStore.
    /// </summary>
    public class CipherStore
    {
        public CipherStore() { }
        public CipherStore(string Filename) { _FileName = Filename; }

        String _FileName = "c:\\CipherText.bin";

        public String FileName
        {
            get { return _FileName; }
            set { _FileName = value; }
        }

        public void Save(CipherData data)
        {
            File.Delete(FileName);
            using (FileStream fs = File.OpenWrite(_FileName))
            {
                fs.WriteByte((byte)data.InitializationVector.Length);
                fs.Write(data.InitializationVector, 0, data.InitializationVector.Length);

                byte[] b = System.Text.Encoding.ASCII.GetBytes(data.Passphrase);
                fs.WriteByte((byte)b.Length);
                fs.Write(b, 0, b.Length);
                
                fs.WriteByte((byte)data.Salt.Length);
                fs.Write(data.Salt, 0, data.Salt.Length);
                
                fs.Write(data.CipherText, 0, data.CipherText.Length);
            }
        }

        public CipherData Load()
        {
            CipherData data = new CipherData();
            using (FileStream fs = File.OpenRead(_FileName))
            {
                byte x = (byte) fs.ReadByte();
                byte[] b = new byte[x];
                fs.Read(b, 0, b.Length);
                data.InitializationVector = b;
                
                x = (byte) fs.ReadByte();
                b = new byte[x];
                fs.Read(b, 0, b.Length);
                data.Passphrase = System.Text.Encoding.ASCII.GetString(b);
                
                x = (byte) fs.ReadByte();
                b = new byte[x];
                fs.Read(b, 0, b.Length);
                data.Salt= b;

                b = new byte[(int)(fs.Length - fs.Position)];
                fs.Read(b, 0, b.Length);
                data.CipherText = b;
            }
            return data;
        }

    }
}

File: samples/AesDemo-Java/examples/CipherStore.java
// CipherStore.java
//
// ------------------------------------------------------------------
//
// This class knows how to load and store an instance of CipherData.
//
// built on host: DINOCH-2
// Created Sat Aug 08 23:11:27 2009
//
// last saved: 
// Time-stamp: <2009-August-09 02:28:33>
// ------------------------------------------------------------------
//
// Copyright (c) 2005-2009 by Dino Chiesa
// All rights reserved!
//
// ------------------------------------------------------------------

package dpchiesa.examples;

import java.io.IOException;
import java.io.FileInputStream;
import java.io.FileOutputStream;


public class CipherStore {
    
    String FileName= "c:\\CipherText.bin";

    //public CipherStore() {}

    public CipherStore(String filename) {
        FileName = filename;
    }
    
    public void Save(CipherData data) throws IOException {
        FileOutputStream fs= new FileOutputStream(FileName);
                
        fs.write((byte)data.InitializationVector.length);
        fs.write(data.InitializationVector);
                
        byte[] b= data.Passphrase.getBytes();
        fs.write((byte)b.length);
        fs.write(b);
                
        fs.write((byte)data.Salt.length);
        fs.write(data.Salt);
                
        fs.write(data.CipherText); 
        fs.close();
    }
        
    public CipherData Load() throws IOException {
        // Read file here
        FileInputStream fs= new FileInputStream(FileName); 
        CipherData data= new CipherData();
        byte x= (byte) fs.read();
        data.InitializationVector = new byte[x];
        fs.read(data.InitializationVector, 0, data.InitializationVector.length); 

        x= (byte) fs.read();
        byte[] b = new byte[x];
        fs.read(b, 0, b.length);
        data.Passphrase = new String(b); // ascii encoding

        x= (byte) fs.read();
        data.Salt = new byte[x];
        fs.read(data.Salt, 0, data.Salt.length); 
                
        int length= fs.available();
        data.CipherText= new byte[length];
        fs.read(data.CipherText, 0, length); 
        fs.close();
                
        return data;
                
    }
}
